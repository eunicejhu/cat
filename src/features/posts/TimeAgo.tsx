import React from "react";
import getTimeAgo from "../../utils/getTimeAgo";

const TimeAgo: React.FC<{ date: string }> = ({ date }) => {
  const timeAgo = getTimeAgo(date);
  const hasValidTimeAgo = Boolean(timeAgo);
  return (
    <div className="date">{`${hasValidTimeAgo ? timeAgo + " ago" : ""}`} </div>
  );
};
export default TimeAgo;
