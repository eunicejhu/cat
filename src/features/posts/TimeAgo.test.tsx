import React from "react";
import TimeAgo from "./TimeAgo";
import renderer from "react-test-renderer";

jest.mock("../../utils/getTimeAgo.ts", () =>
  jest.fn().mockReturnValue("28 minutes")
);
test("render correctly", () => {
  const date = "2020-11-15T16:16:08.493Z";
  const tree = renderer.create(<TimeAgo date={date} />).toJSON();
  expect(tree).toMatchSnapshot();
});
