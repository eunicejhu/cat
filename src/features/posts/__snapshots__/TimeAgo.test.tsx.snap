// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="date"
>
  28 minutes ago
   
</div>
`;
